### ISSUE TITLE
Separate entry names from custom names, or rename thumbnails accordingly.

### ISSUE THREAD

**Issue by engiewb** [libretro/RetroArch#12252]
## Description

When you rename a playlist entry though RetroArch, the thumbnails and information are lost, since RetroArch relies on the entry name for both. A simple solution would be to rename the thumbnail files automatically when the user renames an entry, but I don't know how the information works so I can't say anything on that.

Another would just be to add a custom label that is defined by the user, that doesn't touch anything about the entry itself. I think this is the most elegant option, since then you could name the file and the playlist entry however you want.

RetroArch already generates the entry using the No-Intro name regardless of the file name, meaning it detects exactly what ROM it is. I don't see why the thumbnails and information should rely on the entry name instead of that.

### Expected behavior

The thumbnails and information stay as they are.

### Actual behavior

The thumbnails and information disappear.

### Steps to reproduce the bug

1. Import content.
2. Download thumbnails.
3. Rename the content's entry name.

### Bisect Results

Has always been this way.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: cf6ca2a

### Environment information

- OS: Android 9 and Windows 10 20H2.

**Comment by andiandi13** [comment]
I second this. It's something that bothered me when I tried to change the name of my arcade roms.

Example vf2 = Virtua Fighter 2, but if you rename the zip file, it doesnt work since arcade roms have to have a specific name.

But renaming the entry only unlink thumbnails.

A solution could be to let us choose how Retroarch detect thumbnails :

1) CRC32 detection : whatever the name of the entry or the filename is, RA will check the CRC32 and display the thumbnail corresponding to the rom.

2) Filename detection : whatever the name of the entry or it's CRC32 is, RA will check the name of your rom file to associate a thumbnail

3) Entry detection : will based on entry name, not CRC32 nor filename

4) Automatic : Will check CRC32 first, if no thumbnail linked to database name is detected, will check rom filename, if nothing is detected will check entry name at last.

That's an idea, I don't know if it'd be feasible this way, but something should be implemented to let us rename entries without loosing anything.

**Comment by loucurapt** [comment]
Yes, I think a "display_label" should be added to the playlist files (JSON format).

By default RetroArch would use the "label" value as usual, but renaming a game inside RetroArch would add/modify this "display_label", which would then be displayed in the interface.

Another idea: if a thumbnail with the game's "label" doesn't exist in the thumbnails folder, it could try searching by the file name of the game (minus extension).

**Comment by slacknk** [comment]
It is good idea with `display_label` ( or `label_display` ) in JSON.lpl, Because, if do it now custom name label_name for franchise's separate sometimes don't work  in any system (with .dot in name). Example:
```
    {
      "label": "[Simpsons, The] Bart Simpson's Escape from Camp Deadly (USA, Europe)",
      "path": "Bart Simpson's Escape from Camp Deadly (USA, Europe)",
      "db_name": "Nintendo - Game Boy.lpl"
    },
    {
      "label": "[Simpsons, The] Bart & the Beanstalk (USA, Europe)",
      "path": "Simpsons, The - Bart _ the Beanstalk (USA, Europe)",
      "db_name": "Nintendo - Game Boy.lpl"
    },
    {
      "label": "[Simpsons, The] Bart vs. the Juggernauts (USA, Europe)",
      "path": "Simpsons, The - Bart vs. the Juggernauts (USA, Europe)",
      "db_name": "Nintendo - Game Boy.lpl"
    },
```
```
    {
      "label": "Bart Simpson's Escape from Camp Deadly (USA, Europe)",
      "path": "Bart Simpson's Escape from Camp Deadly (USA, Europe)",
      "db_name": "Nintendo - Game Boy.lpl"
    },
    {
      "label": "Simpsons, The - Bart & the Beanstalk (USA, Europe)",
      "path": "Simpsons, The - Bart _ the Beanstalk (USA, Europe)",
      "db_name": "Nintendo - Game Boy.lpl"
    },
    {
      "label": "Simpsons, The - Bart vs. the Juggernauts (USA, Europe)",
      "path": "Simpsons, The - Bart vs. the Juggernauts (USA, Europe)",
      "db_name": "Nintendo - Game Boy.lpl"
    },
```
*  `"label": "[Simpsons, The] Bart vs. the Juggernauts (USA, Europe)",` - don't work/show, 
* other works fine
